source/javascripts/components/PipelinesPage/providers/SearchParamsProvider.tsx

```diff
@@ -0,0 +1,40 @@
+import { Dispatch, PropsWithChildren, SetStateAction, createContext, useEffect, useMemo, useState } from 'react';
+
+type Props = PropsWithChildren;
+
+type Context = {
+  searchParams: URLSearchParams;
+  setSearchParams: Dispatch<SetStateAction<URLSearchParams>>;
+};
+
+export const SearchParamsContext = createContext<Context>({
+  searchParams: new URLSearchParams(),
+  setSearchParams: () => undefined,
+});
+
+const SearchParamsProvider = ({ children }: Props) => {
+  const [searchParams, setSearchParams] = useState(new URLSearchParams(window.location.hash.split('?')[1] || ''));
```

Comment:
Creating a helper function for this one couldn't hurt, you already write this twice.
`new URLSearchParams(window.location.hash.split('?')[1] || '')`